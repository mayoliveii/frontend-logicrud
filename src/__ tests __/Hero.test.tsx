import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Hero from '../components/Hero/Hero';
import '@testing-library/jest-dom';

jest.mock('./../apiServices/apiService', () => ({
  createCategory: jest.fn(),
}));

describe('Hero', () => {
  it('renders Hero component and matches snapshot', () => {
    const { asFragment } = render(
      <ChakraProvider>
        <Router>
          <Hero />
        </Router>
      </ChakraProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the heading', () => {
    const { getByText } = render(
      <ChakraProvider>
        <Router>
          <Hero />
        </Router>
      </ChakraProvider>
    );
    expect(getByText('Gerenciamento eficiente')).toBeInTheDocument();
    expect(getByText('de veículos de transporte de carga')).toBeInTheDocument();
  });

  it('renders the description', () => {
    const { getByText } = render(
      <ChakraProvider>
        <Router>
          <Hero />
        </Router>
      </ChakraProvider>
    );
    expect(getByText('Gerencie sua frota de transporte com facilidade. Cadastre, atualize e remova veículos com eficiência. Acompanhe manutenção, custos e disponibilidade. Simplifique suas operações logísticas com LogiCRUD.')).toBeInTheDocument();
  });
});