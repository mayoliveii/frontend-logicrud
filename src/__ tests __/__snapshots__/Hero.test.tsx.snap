// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders Hero component and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-cwlxpc"
  >
    <div
      class="chakra-stack css-wqkz5d"
    >
      <div
        class="chakra-stack css-qw9c5c"
      >
        <h2
          class="chakra-heading css-58r1o4"
        >
          <span
            class="chakra-text css-aq3u6h"
          >
            Gerenciamento eficiente
          </span>
          <br />
          <span
            class="chakra-text css-sg8z9j"
          >
            de veículos de transporte de carga
          </span>
        </h2>
        <p
          class="chakra-text css-q9k0mw"
        >
          Gerencie sua frota de transporte com facilidade. Cadastre, atualize e remova veículos com eficiência. Acompanhe manutenção, custos e disponibilidade. Simplifique suas operações logísticas com LogiCRUD.
        </p>
        <div
          class="chakra-stack css-18dwh15"
        >
          <button
            class="chakra-button css-drurm2"
            type="button"
          >
            <a
              href="/vehicle-list"
            >
              Ver veículos
            </a>
          </button>
          <button
            class="chakra-button css-o1afv5"
            type="button"
          >
            Nova categoria
          </button>
        </div>
      </div>
      <div
        class="css-n8pbor"
      >
        <svg
          class="chakra-icon css-18m8396"
          focusable="false"
          viewBox="0 0 578 440"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M239.184 439.443c-55.13-5.419-110.241-21.365-151.074-58.767C42.307 338.722-7.478 282.729.938 221.217c8.433-61.644 78.896-91.048 126.871-130.712 34.337-28.388 70.198-51.348 112.004-66.78C282.34 8.024 325.382-3.369 370.518.904c54.019 5.115 112.774 10.886 150.881 49.482 39.916 40.427 49.421 100.753 53.385 157.402 4.13 59.015 11.255 128.44-30.444 170.44-41.383 41.683-111.6 19.106-169.213 30.663-46.68 9.364-88.56 35.21-135.943 30.551z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="css-k68dew"
        >
          <img
            alt="Hero Image"
            class="chakra-image css-1qtvm2r"
            src="https://mayoliveii.s3.us-east-2.amazonaws.com/images/merchandise-truck.jpg"
          />
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
